src/main/java/domain/LadderResult.java

```diff
@@ -0,0 +1,53 @@
+package domain;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LadderResult {
+
+    private final Map<Position, Position> firstAndLastPosition;
+
+    public LadderResult(Ladder ladder, int participantsCount) {
+        firstAndLastPosition = new LinkedHashMap<>();
+        initPosition(participantsCount);
+        calculatePosition(ladder);
+    }
+
+    private void initPosition(int participantsCount) {
+        for (int i = 0; i < participantsCount; i++) {
+            firstAndLastPosition.put(new Position(i), new Position(i));
+        }
+    }
+
+    private void calculatePosition(Ladder ladder) {
+        for (int floor = 0; !ladder.isFinish(floor); floor++) {
+            moveAllPosition(ladder, floor);
+        }
+    }
+
+    private void moveAllPosition(Ladder ladder, int floor) {
+        for (Map.Entry<Position, Position> entry : firstAndLastPosition.entrySet()) {
+            Position firstPosition = entry.getKey();
+            moveEachPosition(ladder, floor, firstPosition);
+        }
+    }
+
+    private void moveEachPosition(Ladder ladder, int floor, Position firstPosition) {
+        if (ladder.canMoveLeft(floor, firstAndLastPosition.get(firstPosition).getPosition())) {
+            firstAndLastPosition.get(firstPosition).movePositionLeft();
+            return;
+        }
+        if (ladder.canMoveRight(floor, firstAndLastPosition.get(firstPosition).getPosition())) {
+            firstAndLastPosition.get(firstPosition).movePositionRight();
+        }
+    }
+
+    public int getOneResult(int firstPositionValue) {
+        return firstAndLastPosition.entrySet()
+                .stream()
+                .filter(positions -> positions.getKey().getPosition() == firstPositionValue)
+                .mapToInt(positions -> positions.getValue().getPosition())
+                .findFirst()
+                .getAsInt();
+    }
```

Comment:
이 부분은 map 객체의 자료구조 이점을 활용하지 못하고 있습니다!

map은 시간 복잡도가 O(1)인데, map을 list화 하여 O(n) 복잡도로 만들었습니다. (시간복잡도 개념을 아시나요? 모르신다면 두고, Map 자료구조에 대해서만 학습해보세요! list에 비해 탐색에서 엄청난 성능 이득이 있습니다.)

다음처럼 해보는건 어떨까요?
```suggestion
    public int getOneResult(int firstPositionValue) {
        return firstAndLastPosition.get(new Position(firstPositionValue)).getPosition();
    }
```

---
달리는 이렇게도 가능합니다. (이쪽이 제 취향)
```suggestion
    public Position getOneResult(Position first) {
           return firstAndLastPosition.get(first);
    }
```

Reply:
아래 구조가 더 깔끔한 것 같아 변경하였으나 getter 사용이 일부 증가하여 고민하였습니다
다만 트레이드 오프임을 인지하고 꼭 필요한 로직이라고 판단한 바 그대로 진행하였습니다!
Height내에 내부 로직을 만들어 진행할 수 있는 건 그렇게 진행하였습니다